pyartcd/pyartcd/pipelines/build_microshift.py

```diff
@@ -127,17 +127,13 @@ async def run(self):
                     pr = await self._create_or_update_pull_request(nvrs)
 
             # Sends a slack message
-            message = f"Hi @release-artists , microshift for assembly {self.assembly} has been successfully built."
+            message = f"microshift for assembly {self.assembly} has been successfully built."
             if pr:
                 message += f"\nA PR to update the assembly definition has been created/updated: {pr.html_url}"
-                message += "\nReview and merge the PR before you proceed.\n"
-            message += '\n'.join([
-                "To attach the build to Errata, run <https://saml.buildvm.hosts.prod.psi.bos.redhat.com:8888/job/aos-cd-builds/job/build%252Fprepare-release/|prepare-release> job, or:",
-                "```",
-                f"elliott --group {self.group} --assembly {self.assembly} --rpms microshift find-builds -k rpm --member-only --use-default-advisory microshift",
-                f"elliott --group {self.group} --assembly {self.assembly} attach-cve-flaws --use-default-advisory microshift",
-                "```"
-            ])
+                message += f"\n@release-artists Attach build to microshift advisory failed" if await self._attach_builds() !=0 else f"\nAttach build to microshift advisory succeed"
+                message += f"\n@release-artists Attach cve flaws to microshift advisory failed" if await self._attach_cve_flaws() !=0 else f"\nAttach cve flaws to microshift advisory succeed"
+                message += f"\n@release-artists Verify microshift bugs failed" if await self._verify_microshift_bugs() !=0 else f"\nVerify microshift bugs succeed"
+                message += f"\n@release-artists Change microshift advisory status to QE failed" if await self._change_advisory_status() !=0 else f"\nChange microshift advisory status to QE succeed"
```

Comment:
If any of _attach_builds, _attach_cve_flaws, _verify_microshift_bugs, or _change_advisory_status fails, the job still succeeds. I think it will make more sense to fail the job then tag release-artist to have a look.

Reply:
I think those steps are not major part of building microshift job, and those parts eventually can move to promote job(but currently pre-steps in promote job haven't migrate to python yet) maybe more reasonable to fail promote job.As the following except treat error says "Error building microshift", only failed brew build or failed to create pr is the real status for the job.

Reply:
Makes sense. But could you update those functions to let them raise an exception on failure? Then from the caller you `try-except` them and send those messages in the `except` block.
Returning 0 or 1 from a python function for exception handling is like an anti-pattern.

Reply:
The slack messaging can be improved to something like:
- After build completes, send a message about the success without a ping
- Before preparing the advisory, send another message without a ping, something like `preparing microshift advisory...`.
- If bugs and builds are attached without any error, send another message like `advisory has been prepared: {errata_link}`.
- If failed, send a message like `Error preparing microshift advisory: {error messsage}. \n Please have a look and use the following commands to do it manually: {commands}`